packages/slice-machine/src/legacy/lib/builders/CustomTypeBuilder/SliceZone/index.tsx

```diff
@@ -429,6 +433,38 @@ const SliceZone: React.FC<SliceZoneProps> = ({
       )}
       <GenerateSliceWithAiModal
         open={isGenerateSliceWithAiModalOpen}
+        onSuccess={async (slices: SharedSlice[]) => {
+          const serverState = await getState();
+          // Update Redux store
+          createSliceSuccess(serverState.libraries);
+
+          const newCustomType = addSlicesToSliceZone({
+            customType,
+            tabId,
+            slices,
+          });
+          setCustomType(CustomTypes.fromSM(newCustomType), () => {
+            toast.success(
+              <ToastMessageWithPath
+                message="New slices added to slice zone and created at: "
+                path={`${localLibraries[0].name}/`}
```

Comment:
this and line 461 is true only in practice (becuase it happens that we use the first/default library to create the slices to within the modal, but these two parts of the code are completely disconnected

we should either pass to the modal the library we want to use, or the on success callback should pass here the library it used, that way a change in one will be connected with the other

Reply:
Alright, the library has one source of truth now ☑️